File: Python/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0

        res = 0

        for i in range(1, len(prices)):
            if prices[i] > prices[i - 1]:
                res += prices[i] - prices[i - 1]
        
        return res

File: Swift/best-time-to-buy-and-sell-stock-ii.swift
class Solution {
  func maxProfit(_ prices: [Int]) -> Int {
    guard prices.count >= 2 else {
      return 0
    }
    
    var res = 0
    
    for i in 1..<prices.count where prices[i] > prices[i - 1] {
      res += prices[i] - prices[i - 1]
    }
    
    return res
  }
}
